File: edukate-backend/src/main/java/io/github/sanyavertolet/edukate/backend/configs/RabbitConfig.java
package io.github.sanyavertolet.edukate.backend.configs;

import io.github.sanyavertolet.edukate.messaging.RabbitTopology;
import org.springframework.amqp.core.*;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class RabbitConfig {

    @Bean
    public Declarables resultTopology(TopicExchange edukateExchange) {
        Queue resultQueue = QueueBuilder.durable(RabbitTopology.Q.RESULT_BACKEND).build();
        Binding binding = BindingBuilder.bind(resultQueue).to(edukateExchange).with(RabbitTopology.Rk.RESULT);
        return new Declarables(resultQueue, binding);
    }
}

File: edukate-backend/src/main/kotlin/io/github/sanyavertolet/edukate/backend/configs/RabbitConfig.kt
package io.github.sanyavertolet.edukate.backend.configs

import io.github.sanyavertolet.edukate.messaging.RabbitTopology
import org.springframework.amqp.core.Binding
import org.springframework.amqp.core.BindingBuilder
import org.springframework.amqp.core.Declarables
import org.springframework.amqp.core.QueueBuilder
import org.springframework.amqp.core.TopicExchange
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class RabbitConfig {
    @Bean
    fun resultTopology(edukateExchange: TopicExchange): Declarables {
        val resultQueue = QueueBuilder.durable(RabbitTopology.Q.RESULT_BACKEND).build()
        val binding: Binding = BindingBuilder.bind(resultQueue).to(edukateExchange).with(RabbitTopology.Rk.RESULT)
        return Declarables(resultQueue, binding)
    }
}
